lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestDisposeTasks.cs

```diff
@@ -167,20 +168,16 @@ public void OnNext(ICompletedTask value)
             /// And verify the context associated with the failed task is the same as the context that the task was submitted
             /// </summary>
             /// <param name="value"></param>
-            public void OnNext(IFailedTask value)
+            public void OnNext(IFailedEvaluator value)
             {
-                Assert.Equal(TaskId + "2", value.Id);
+                Assert.True(value.FailedTask.IsPresent());
+                Assert.Equal(TaskId + "2", value.FailedTask.Value.Id);
 
-                var failedException = ByteUtilities.ByteArraysToString(value.Data.Value);
-                var e = value.AsError();
+                var e = value.EvaluatorException.InnerException;
                 Logger.Log(Level.Error, "In IFailedTask: e.type: {0}, e.message: {1}.", e.GetType(), e.Message);
-                Logger.Log(Level.Error, "In IFailedTask: value.Data.Value: {0}, value.Message {1}.", failedException, value.Message);
 
                 Assert.Equal(typeof(Exception), e.GetType());
```

Comment:
May be throw a special exception in the test task, like ApplicationException, so that when checking here, we know it is now a generic exception.